x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/layer_panel.tsx

```diff
@@ -297,26 +303,66 @@ export function LayerPanel(
                         }}
                       >
                         <div className="lnsLayerPanel__dimension">
-                          <NativeRenderer
-                            render={props.datasourceMap[datasourceId].renderDimensionTrigger}
-                            nativeProps={{
-                              ...layerDatasourceConfigProps,
-                              columnId: accessor,
-                              filterOperations: group.filterOperations,
-                              suggestedPriority: group.suggestedPriority,
-                              onClick: () => {
-                                if (activeId) {
-                                  setActiveDimension(initialActiveDimensionState);
-                                } else {
-                                  setActiveDimension({
-                                    isNew: false,
-                                    activeGroup: group,
-                                    activeId: accessor,
-                                  });
-                                }
-                              },
+                          <EuiLink
+                            className="lnsLayerPanel__dimensionLink"
+                            onClick={() => {
+                              if (activeId) {
+                                setActiveDimension(initialActiveDimensionState);
+                              } else {
+                                setActiveDimension({
+                                  isNew: false,
+                                  activeGroup: group,
+                                  activeId: accessor,
+                                });
+                              }
                             }}
-                          />
+                          >
+                            <EuiFlexGroup gutterSize="none" alignItems="center">
+                              {typeof accessorConfig !== 'string' &&
```

Comment:
Could we move this logic to separate component that takes `triggerIcon` and `color` props, eg. <ColorIndicator />?